### ISSUE TITLE
If you specify `3.5.1` it will install `3.5.10`

### ISSUE THREAD

**Issue by slifty** [momo-lab/pyenv-install-latest#6]
This is due to the way version strings are being compared -- it should instead compare against either a perfect match or a `QUERY.*` instead of `QUERY*`.

<img width="482" alt="image" src="https://user-images.githubusercontent.com/208884/148336686-3a07a5d8-bef5-45cf-8926-f1ad7bd8c395.png">

**Comment by slifty** [comment]
Also notice the whitespace between "Install" and the version -- it should trim whitespace ideally!

I've fixed this locally so I'll open a PR for your consideration

**Comment by slifty** [comment]
PR opened: https://github.com/momo-lab/pyenv-install-latest/pull/7

**Comment by momo-lab** [comment]
I think the following will solve the problem. I'll try it later.

```diff
-    | grep -E "^\s*$query" \
+    | grep -E "^\s*${query/./\\.}\b" \
```

**Comment by momo-lab** [comment]
Fix it in #8, please give it a try.

Also, pyenv-install-latest has been deplicated, so please migrate to xxenv-latest.
The same fix is applied.

**Comment by slifty** [comment]
Thank you!
